frontend/src/components/datasets/ChooseLicenseModal.tsx

```diff
@@ -0,0 +1,109 @@
+import React, { useEffect, useState } from "react";
+import {
+	Box,
+	Button, Dialog,
+	DialogContent,
+	DialogTitle,
+	FormControl,
+	Link
+} from "@mui/material";
+import CircleIcon from '@mui/icons-material/Circle';
+import { V2 } from "../../openapi";
+import { LicenseOption } from "../../openapi/v2";
+import {CreateLicenseModal} from "./CreateLicenseModal";
+
+type ChooseLicenseModalProps = {
+	selectedLicense: any;
+  setSelectedLicense: any;
+  setLicenseRequestForm: any;
+  handleBack: any;
+  handleNext: any
+};
+
+export const ChooseLicenseModal: React.FC<ChooseLicenseModalProps> = (
+	props: ChooseLicenseModalProps
+) => {
+	const { selectedLicense, setSelectedLicense, setLicenseRequestForm, handleBack, handleNext } = props;
+	const [standardLicenses, setStandardLicenses] = useState<LicenseOption[]>([]);
+	const [licenseModalOpen, setLicenseModalOpen] = useState<boolean>(false);
+
+	useEffect(() => {
+		V2.LicensesService.getLicensesApiV2LicensesGet()
```

Comment:
You can put this into redux/action;
If you think it's overkill, could you put it under util? If the backend endponit changed, this auto generated function name also changed. It would be very hard to trace it if it's embedded in the component like this